### ISSUE TITLE
build error

### ISSUE THREAD

**Issue by myungjin** [libreswan/libreswan#348]
I am having compilation error for commit version 3cc204cb1670c5efe0946a3134623d09f465b4e6.

```
/work/libreswan/programs/pluto/log.c: In function ‘cur_logger’:
/work/libreswan/programs/pluto/log.c:985:4: error: initializer element is not constant
    .where = { .func = __func__, .basename = HERE_BASENAME , .line = __LINE__},/*HERE*/
    ^
/work/libreswan/programs/pluto/log.c:985:4: error: (near initialization for ‘pexpect_logger.where.basename’)
```

I was able to compile commit version 044ed08e3f945de3519af452291e50586d58daa8 on the same environment of mine.

I am using Centos 7. gcc version is 4.8.5.

Is there any stable/recommended setting for compiling a development version? Please let me know.

**Comment by myungjin** [comment]
FYI, I can confirm that version to commit a73bd43996 is compiled successfully; but beginning from commit cf0527ab58, the build fails.

**Comment by cagney** [comment]
Try 0ac878b8e77f3c47d7910be4d225c8b7fc773c81

**Comment by myungjin** [comment]
Passed the error, but hit a new error:
```
	-c /work/libreswan/programs/pluto/iface_udp.c
In file included from ../../include/refcnt.h:22:0,
                 from /work/libreswan/programs/pluto/iface.h:26,
                 from /work/libreswan/programs/pluto/iface_udp.c:46:
/work/libreswan/programs/pluto/iface_udp.c: In function ‘udp_read_packet’:
/work/libreswan/programs/pluto/log.h:187:20: error: the comparison will always evaluate as ‘true’ for the address of ‘stack_md’ will never be NULL [-Werror=address]
   if (pexpect((MD) != NULL) &&    \
                    ^
../../include/lswlog.h:492:22: note: in definition of macro ‘pexpect’
   bool assertion__ = ASSERTION;    \
                      ^
/work/libreswan/programs/pluto/iface_udp.c:315:3: note: in expansion of macro ‘log_md’
   log_md(RC_LOG, &stack_md, "recvfrom on %s failed "PRI_ERRNO,
   ^
/work/libreswan/programs/pluto/log.h:187:20: error: the comparison will always evaluate as ‘true’ for the address of ‘stack_md’ will never be NULL [-Werror=address]
   if (pexpect((MD) != NULL) &&    \
                    ^
../../include/lswlog.h:492:22: note: in definition of macro ‘pexpect’
   bool assertion__ = ASSERTION;    \
                      ^
/work/libreswan/programs/pluto/iface_udp.c:324:4: note: in expansion of macro ‘log_md’
    log_md(RC_LOG, &stack_md, "too small packet (%zd)",
    ^
/work/libreswan/programs/pluto/log.h:187:20: error: the comparison will always evaluate as ‘true’ for the address of ‘stack_md’ will never be NULL [-Werror=address]
   if (pexpect((MD) != NULL) &&    \
                    ^
../../include/lswlog.h:492:22: note: in definition of macro ‘pexpect’
   bool assertion__ = ASSERTION;    \
                      ^
/work/libreswan/programs/pluto/iface_udp.c:330:4: note: in expansion of macro ‘log_md’
    log_md(RC_LOG, &stack_md, "has no Non-ESP marker");
    ^
/work/libreswan/programs/pluto/log.h:187:20: error: the comparison will always evaluate as ‘true’ for the address of ‘stack_md’ will never be NULL [-Werror=address]
   if (pexpect((MD) != NULL) &&    \
                    ^
../../include/lswlog.h:492:22: note: in definition of macro ‘pexpect’
   bool assertion__ = ASSERTION;    \
                      ^
/work/libreswan/programs/pluto/iface_udp.c:348:4: note: in expansion of macro ‘log_md’
    log_md(RC_LOG, &stack_md, "mangled with potential spurious non-esp marker");
    ^
cc1: all warnings being treated as errors
make[3]: *** [iface_udp.o] Error 1
make[2]: *** [base] Error 2
make[2]: Leaving directory `/work/libreswan/programs/pluto'
make[1]: *** [recursive-base] Error 2
make[1]: Leaving directory `/work/libreswan/programs'
make: *** [recursive-base] Error 2
```

**Comment by cagney** [comment]
that was fixed with a series of commits, last being f15632b2ee42f41b97893f8ef0ed53f698ec3884

**Comment by myungjin** [comment]
Thank you. I was able to compile the source code.
